src/sage/groups/finitely_presented.py

```diff
@@ -1532,7 +1638,184 @@ def alexander_matrix(self, im_gens=None):
         rel = self.relations()
         gen = self._free_group.gens()
         return matrix(len(rel), len(gen),
-                      lambda i,j: rel[i].fox_derivative(gen[j], im_gens))
+                      lambda i, j: rel[i].fox_derivative(gen[j], im_gens))
+
+    def abelian_alexander_matrix(self, ring=QQ, abelianized=None, simplified=True):
+        """
+        Return the Alexander matrix of the group with values in the group
+        algebra of the abelianized.
+
+        INPUT:
+
+        - ``ring`` -- (default: ``QQ``) the base ring of the
+          group algebra
+        - ``abelianized`` -- optional (default: ``None``); a tuple containing the output of
+          ``self.abelianization_to_algebra()`` to avoid recomputation
+        - ``simplified`` -- boolean (default: ``False``); if set to
+          ``True`` use Gauss elimination and erase rows and columns
+
+        OUTPUT:
+
+        - ``A`` -- a matrix with coefficients in ``R``
+
+        - ``R`` -- a ring of Laurent polynomials over ``ring``
+
+        - ``ideal`` -- an list of generators of an ideal ``I`` of ``R`` such that ``R/I`` is
+          the group algebra of the abelianization of ``self``
```

Comment:
```suggestion
        - ``A`` -- a matrix with coefficients in ``R``
```
This should only return the matrix `A`. The ring `R` can be obtained from that and the ideal is not anything (effectively) created in this computation.

Reply:
The ring is no more returned but the ideal is; after testing I checked that the other necessary changes in the tests and code were done.